assembl/views/search/views.py

```diff
@@ -0,0 +1,62 @@
+import json
+from pyramid.events import NewRequest
+from pyramid.view import view_config
+
+from assembl.indexing.utils import connect
+from assembl.indexing.settings import get_index_settings
+from assembl.indexing.changes import changes, in_tests
+from assembl import models
+from assembl.lib.sqla import get_session_maker
+
+def get_curl_query(query):
+    return "curl -XGET 'localhost:9200/_search?pretty' -d '{}'".format(
+        json.dumps(query).replace("'", "\\u0027"))
+
+
+@view_config(route_name='search', renderer='json')
+def search_endpoint(context, request):
+    es = connect()
+    index_name = get_index_settings()['index_name']
+    query = request.json_body
+#    print get_curl_query(query)
+    result = es.search(index=index_name, body=query)
+
+    # add creator_name in each hit
+    creator_ids = set([hit['_source']['creator_id']
+                       for hit in result['hits']['hits']
+                       if hit['_source'].get('creator_id', None) is not None])
+    session = get_session_maker()
+    creators = session.query(models.AgentProfile.id, models.AgentProfile.name
+        ).filter(models.AgentProfile.id.in_(creator_ids)).all()
+    creators_by_id = dict(creators)
+    # u'query': {u'bool': {u'filter': [{u'term': {u'discussion_id': u'23'}}]}}
+    filters = [fil for fil in query['query']['bool']['filter']]
+    discussion_id = [f.values()[0].values()[0]
```

Comment:
Are you sure this is enough to prevent elasticsearch from leaking private discussion data? I'm talking about people who could use this route and craft elasticsearch queries. We need to be really sure of this. I don't know Elasticsearch query language well enough to be sure that this code is secure enough. But this code transfers security concern from our current system to ES query language, which makes me feel unconfortable as the user can craft send any complex ES query (for example, can a user find a way to override our discussion_id filter, or to make multiple queries into one, where only one would have this filter?). I guess the other way would be that our backend does the ES search and then parses each hit to check if user is allowed to read this hit, but this is far from ideal. Maybe we should look a bit more into ES security modules (ACLs, etc for example via X-Pack/Shield?).